flag_engine/identities/models.py

```diff
@@ -1,28 +1,65 @@
 import datetime
 import typing
 import uuid
-from dataclasses import dataclass, field
 
+from pydantic import UUID4, BaseModel, Field, root_validator
+from pydantic_collections import BaseCollectionModel
+
+from flag_engine.features.models import FeatureStateModel
 from flag_engine.identities.traits.models import TraitModel
-from flag_engine.utils.collections import IdentityFeaturesList
 from flag_engine.utils.datetime import utcnow_with_tz
+from flag_engine.utils.exceptions import DuplicateFeatureState
+
+
+class IdentityFeaturesList(BaseCollectionModel[FeatureStateModel]):
+    @classmethod
+    def __get_validators__(
```

Comment:
Just a general comment here that I definitely prefer the `marshmallow` `@validates_schema` decorator for providing custom decorators! Maybe this is more explicit though.

Reply:
After switching to Pydantic v2 we'll be able to use a custom `List` subclass instead of this hacky collection model.

Or we could (something I'd prefer to do) not use custom collections altogether and move their validation/interfaces to the parent model.

Reply:
Done.